Add entrys to steps list for uncaught exceptions

timbrown5 opened the pull request:
Some failed steps were getting missing.
Check for ErrorAction in nodes and if present add the node. This will contain some error text and (possibly) an exception.

This is enough for actions that return an AbortException, and to fix #213, but wouldn't include errors like: https://github.com/jenkinsci/pipeline-graph-view-plugin/issues/213#issuecomment-1438666811

To do that we:
Check if the last node (the first one we come to) has an ErrorAction - if so there is an uncaught exception, then:
- Find and record the originating step.
- Check each atomNode and endNode as we go to see if it's the originating node.
-- Once we find the originating step we add that to the `stageSteps` list - it will be mapped to the stage when we hit the start node.

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira

timbrown5 commented:
Manual Tests:
```
stage('failure') {
    echo "This will work"
    echo "THis won't - calling an undefined variable ${undefined}"
}
```
Screenshot:
<img width="974" alt="Screenshot 2023-02-22 at 09 47 34" src="https://user-images.githubusercontent.com/4447764/220756689-fe34dddd-bb78-4e4e-9dba-ac0271861337.png">

Pipeline from https://github.com/jenkinsci/pipeline-graph-view-plugin/issues/213#issuecomment-1438322664

Screenshot:
<img width="1762" alt="Screenshot 2023-02-22 at 20 52 58" src="https://user-images.githubusercontent.com/4447764/220756838-23db5791-c6f3-4b0b-a728-7facc3657a29.png">

timja commented on src/main/java/io/jenkins/plugins/pipelinegraphview/consoleview/PipelineConsoleViewAction.java:
```diff
@@ -103,46 +101,59 @@ public HttpResponse getConsolOutput(StaplerRequest req) throws IOException {
+          }
         }
+
+        logger.info(
```
shouldn't this be debug?

timja commented on src/main/java/io/jenkins/plugins/pipelinegraphview/consoleview/PipelineConsoleViewAction.java:
```diff
@@ -103,46 +101,59 @@ public HttpResponse getConsolOutput(StaplerRequest req) throws IOException {
+        logger.info(
+            "Returning '"
+                + Long.toString(textLength - startByte)
+                + "' bytes from 'getConsolOutput'.");
```
```suggestion
                + "' bytes from 'getConsoleOutput'.");
```

timja commented:
spotless is failing

timbrown5 commented:
> spotless is failing

I've been fighting spotbugs and spotless. Not to self, `mvn clean test spotless:apply spotbugs:check` seems to be working better than running them individually.

timja commented:
> > spotless is failing
> 
> I've been fighting spotbugs and spotless. Not to self, `mvn clean test spotless:apply spotbugs:check` seems to be working better than running them individually.

Weird, I've always just run:
```
mvn spotless:apply
```
and never had an issue with it

timbrown5 commented:
Added two small changes to the look of the console log cards:
Fixed the alignment with the stage details card.
![Screenshot 2023-02-23 at 10 07 38](https://user-images.githubusercontent.com/4447764/220879123-4e7d7ea2-9426-4b1f-8ce7-c8554e23adc7.png)

Changed the expand icon color back:
![Screenshot 2023-02-23 at 10 16 02](https://user-images.githubusercontent.com/4447764/220879303-adb6237d-13fb-43f4-bd81-5d0bb94f6f9d.png)
(I broke this in https://github.com/jenkinsci/pipeline-graph-view-plugin/commit/61fcf9a7fbe7d36af1ace87c35d8ce0a5c5c12c5 - turns out it wasn't unused.

timbrown5 commented:
Also fixed the text-overflow in card headers:
Before:
![Screenshot 2023-02-23 at 11 20 05](https://user-images.githubusercontent.com/4447764/220892172-13f6d4d9-5397-43f5-8a13-394ec1824bbe.png)

After:
![Screenshot 2023-02-23 at 11 18 35](https://user-images.githubusercontent.com/4447764/220892198-85da441c-6b13-4b8d-a497-18b2eae72256.png)

(I was a little too eager with my cleanup before).

timbrown5 commented:
I think the code I added to fix spotbugs issues failed spotless, running
both means I don't miss something 😅

On Thu, 23 Feb 2023, 10:13 Tim Jacomb, ***@***.***> wrote:

> spotless is failing
>
> I've been fighting spotbugs and spotless. Not to self, mvn clean test
> spotless:apply spotbugs:check seems to be working better than running
> them individually.
>
> Weird, I've always just run:
>
> mvn spotless:apply
>
> and never had an issue with it
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/jenkinsci/pipeline-graph-view-plugin/pull/217#issuecomment-1441500404>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABB54FHW4TE5ELCZBBZRLWLWY4Z6FANCNFSM6AAAAAAVDSIAJQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
